src/subscription_manager/api.py

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright (c) 2015 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+
+"""
+Class and methods in this module are provided for use by external applications with
+the exception of methods beginning with an '_'.
+
+All reasonable efforts will be made to maintain compatibility.
+"""
+import fnmatch
+
+from subscription_manager import logutil
+logutil.init_logger()
```

Comment:
> Could this init/setup stuff move into a method instead of module scope?
> 
> If our yum plugins (plugins/subscription-manager.py for ex) import 'api', they would do some of this twice.

Normally, I do not like doing things at module scope, but I thought it was warranted in this case.  Anyone using the API methods is going to want to have initial injection handled for them.

I could add a decorator to the API functions and then have a decorator method that invokes all the injection.  That would at least allow for idempotent imports.  How does that sound?